src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,74 @@
 
 export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
 
-      <Loader />
+  useEffect(() => {
+    const fetchUser = async () => {
+      if (currentTodo) {
+        const userFromServer = await getUser(currentTodo?.userId);
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+        setUser(userFromServer);
+        setIsLoading(false);
+      }
+    };
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+    try {
+      fetchUser();
+    } catch {
+      setUser(null);
+    }
+  }, []);
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">fugiat veniam minus</p>
+  return (
+    <div className="modal is-active" data-cy="modal">
+      <div className="modal-background" />
+      {isLoading
+        ? <Loader />
+        : (
+          <div className="modal-card">
+            <header className="modal-card-head">
+              <div
+                className="modal-card-title has-text-weight-medium"
+                data-cy="modal-header"
+              >
+                {`Todo #${currentTodo?.id}`}
+              </div>
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+              <button
+                type="button"
+                className="delete"
+                data-cy="modal-close"
+                aria-label="modal-close"
+                onClick={() => dispatch(currentTodoActions.removeTodo())}
+              />
+            </header>
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
-        </div>
-      </div>
+            <div className="modal-card-body">
+              <p className="block" data-cy="modal-title">
+                {currentTodo?.title}
+              </p>
+
+              <p className="block" data-cy="modal-user">
+                {currentTodo?.completed ? (
+                  <strong className="has-text-success">Done</strong>
+                ) : (
+                  <strong className="has-text-danger">Planned</strong>
+                )}
+                {' by '}
+                <a href={`mailto:${user?.email}`}>{user?.name}</a>
```

Comment:
```suggestion
               {user && (
                 <>
                  {' by '}
                  <a href={`mailto:${user.email}`}>{user.name}</a>                 
                 </>
               )}
```
In case of user === null